### DISCUSSION TITLE
Status of Qt support in SCons

### DISCUSSION THREAD

**Discussion by mwichmann** [SCons/scons#4018] [General]
At the moment SCons has a tool named `qt` which refers to Qt 3.  The last Qt 3 release was 3.3 in 2004, and it's been effectively discontinued for quite a long time now.  There's no Linux distro I know of where Qt 3 is considered current; the Red Hat distros do ship qt3 packages mainly because they're required by LSB, and they haven't dropped the LSB support packages yet.  You can't get them at all for several years now on Ubuntu.

Qt4 and Qt5 have code available outside the SCons core repository - Dirk Baechle has hosted these in his github, and Qt4 has been pulled into https://github.com/SCons/scons-contrib (a PR is pending for Qt5).  Qt 4 is also pretty dead, the last, LTS, version came out in 2011 (4.8.7 was the last minor version).  Qt 5 is more current, but there's now a Qt6.

What should we do with Qt support in SCons?

Should qt (3) be deprecated/dropped?

Is there a compelling reason to have Qt support in SCons itself, as opposed to an add-on? Advantages: there's no extra install, and tests get run; Disadvantages: someone has to pay attention if something breaks.

Is anyone interested enough to work up a Qt 6 tool?

**Comment by merlin66** [comment]
I find it unlikely that anyone would be using SCons with qt3 today. So deprecating qt3 support (and not supporting qt4) is probably fine. Those who want it could easily maintain their own version of the tool anyway. At my company we moved on from qt3 to qt4 in 2009, and from qt4 to qt5 in 2016. Not yet moved to qt6. We've maintained our own variant of the tool from the start, which has its benefits. It almost never breaks and requires only minimal maintenance.

I would vote for putting the best version out there into SCons itself, to avoid having to install extra packages. Getting a decent version in SCons might just spark enough interest and trigger further contributions. Should not cause a lot of maintenance work.

Hard to predict if people with actually switch over from what they are using today though.

**Reply by bdbaddog** [reply]
Any chance you can contribute your qt5 tool? (is it general enough)
Is it very different from the above referenced qt5 tool in scons-contrib?

**Reply by merlin66** [reply]
I think our tool is the original qt3 version with minimal changes/hacks to make it work with qt4, and not much was done for qt5. So the scons-contrib one is probably a better version to use. I can have a look and see if there's anything useful in our version not present in the contrib one.

**Reply by merlin66** [reply]
I have checked now, and there isn't anything useful in our version of the tool, it is mostly identical to the original SCons qt tool.

We only use the tool for moc'ing and compiling .ui files. Locating include paths, library paths and linking to libraries is handled explicitly by our build system, independently of the qt tool, so we didn't do anything to that part.
